### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by belmeetmule**
### In this PR, I have completed the following activities:

- [ ] Fetch book list from Bookstore API on Books page load using fetch API.
- [ ] Refactored add book and remove book features to persist changes in the server
- [ ] Make use of createasyncthunk

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import { configureStore } from '@reduxjs/toolkit';
+import { applyMiddleware, configureStore } from '@reduxjs/toolkit';
+import thunk from 'redux-thunk';
 import booksReducer from './books/books';
 import categoriesReducer from './categories/categories';
```
Comment (rloterh) [inline src/redux/configureStore.js:4]: - [ ] This activity requires that you use `createAsyncThunk` for creating _thunk action creators_, which should be similar to this example:

```javascript
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
```

File: package.json [hunk 2]
```diff
@@ -7,12 +7,15 @@
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
+    "axios": "^1.1.0",
```
Comment (rloterh) [inline package.json:10]: - [ ] OPTIONAL
Kindly endeavor to remove unused package to keep your external modules compact.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Mulugeta, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Nice job for properly using fetch for API calls

## Required Changes ♻️
_Kindly check the comments under the review._

**You can also consider:**
- I think you should change the header title _Authors_ to _Author_ in your ReadMe, as this project is worked on by sole contributor.
-  It's recommended that you add an MIT license, or any other similar license to your ReadMe. 
Implementing the above change(s) will make your ReadMe appear more professional.


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/configureStore.js (hunk 1 above)
Comment (BenMukebo) [reply src/redux/configureStore.js:4]: > * [ ]  This activity requires that you use `createAsyncThunk` for creating _thunk action creators_, which should be similar to this example:
> 
> ```js
> import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
> import axios from 'axios'; 
> 
> 
> export const getBooks = createAsyncThunk(
>   'books/getBooks',
>   async () => {
>     const response = await axios.get(
>       'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
>     );
>     return response.data;
>   },
> ); 
> ```

- [ ] Kindly as requested by the previous reviewer 🙃 the `createAsyncThunk` must be used in the project for creating thunk action creators, Kindly address this 👍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @belmeetmule, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️


 ## Required Changes ♻️
_Check the comments under the review._

You have submitted a project, that Linters check are not running in the GitHub/action
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @belmeetmule 🤝 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
- [ ] The linters are failing, would you address the linters issue please ? 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ugboaja-Uchechi — APPROVED** [review]
Hi @belmeetmule , 

## Status: APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Linters are passing ✔️ 
- README is descriptive 👍 
- Great job with your design and functionalities 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**